_posts/2015-05-29-chrome-push-api.md

```diff
@@ -0,0 +1,98 @@
+
+#### Registering
+Now that you know what to expect with setting up a suitable environment for your service worker let’s get it installed.
+
+Now, a good installation script should also consider that their might be a service worker already in place and running. A lot of other examples want to go straight to registering a service worker, but that will give you a bunch of redundant service workers. To check for that let’s do the following:
+{% gist 81e37636d6b0a55788f1 check-sw-reg.js %}
+Now that we have the installation status we know if we need to register a service worker or not. So let's say we do. Then we do this:
+
+{% gist 81e37636d6b0a55788f1 reg-sw.js %}
+
+**ServiceWorkerRegistrations** are important to push because they hold the key to our next step which is the **PushManager**. Once we have a reliable way to either register a ServiceWorker or retrieve an existing ServiceWorker's registration, we can continue to...
+
+### The PushManager
+The **PushManager** is a creatively named API that manages our push subscriptions. The entry point for it can be found on that `ServiceWorkerRegistration` I mentioned above. You should take the same approach with this as with service worker registrations. The functions you should know about are: `getSubscription` and `subscribe`.
+{% gist 81e37636d6b0a55788f1 pm-get-sub.js %}
+As you can see here, we check for an existing push subscription, then if one is not found we attempt to get permission for one. This attempt is when the user is prompted to approve notifications.
+
+### Putting It Together
+So, putting the two together, you probably end up with something like this:
+
+{% gist 81e37636d6b0a55788f1 sw-pm-together.js %}
+
+This will auto-register a service-worker and will subscribe your website to push notifications and send the push subscription to your server.
+
+#### PushSubscription
+Providing that everything went well, and the user agreed to be notified, you will be given a **PushSubscription** object that will contain details about how to execute a push notification. ([PushSubscription](http://www.w3.org/TR/push-api/#idl-def-PushSubscription)) This object contains an `endpoint` attribute which is the entry point for push notifications. (The Push API is still in flux, but I would recommend keeping the `endpoint`; Google is the only one running a W3C compliant push server but I am sure there will be more to come and this will be what distinguishes them.)
+
+What will distinguish this user is the `subscriptionId`, this is like a device ID of sorts and will be how you reach this user in particular with your push notifications.
```

Comment:
Consider rephrase: "What distinguishes this user is the `subscriptionId`, which is like a device ID and will be how you reach this particular user with your push notifications.".